scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -0,0 +1,465 @@
+
+In addition to gene expression data, this data set includes data from antibody-derived tags (ADTs) which are used to mark cell-surface proteins. 
+
+## Set Up
+
+To start, we will load some of the libraries we will need later, and set a random number seed for reproducibility.
+
+```{r setup}
+# Load libraries
+library(magrittr) # the pipe (%>%) operator
+library(ggplot2) # plotting functions
+library(SingleCellExperiment) # our main class for storing Single Cell data
+
+
+# Setting the seed for reproducibility
+set.seed(12345)
+```
+
+
+### Input and output
+
+```{r}
+### Input files
+data_dir <- file.path("data", "PBMC-TotalSeqB", "normalized")
+sce_file <- file.path(data_dir, "PBMC_TotalSeqB_normalized_sce.rds")
+
+### output files
+analysis_dir <- file.path("analysis", "PBMC-TotalSeqB")
+if (!dir.exists(analysis_dir)){
+  dir.create(analysis_dir, recursive = TRUE)
+}
+
+cellinfo_file <- file.path(analysis_dir, "PBMC_TotalSeqB_cellinfo.tsv")
+```
+
+
+Read in the SingleCellExperiment
+```{r}
+sce <- readRDS(sce_file)
+```
+
+
+Let's explore what is in the SCE already:
+
+```{r}
+sce
+```
+
+Notice the `altExpNames`?
+Those are the ADT data, stored as an _Alternative Experiment_.
+To access these, we can use the `altExp()` function. 
+Let's look at what we have in that slot:
+
+```{r}
+altExp(sce, "ADT")
+```
+
+
+Ah! Another SingleCellExperiment! Inception!
+Since there is only one `altExp`, the second argument is optional, so we won't use it for the rest of this notebook.
+Here the columns are again the cells, but now the rows are the proteins that were assayed with the antibodies and tags.
+
+```{r}
+# What proteins were assayed?
+adt_tags <- rownames(altExp(sce))
+adt_tags
+```
+
+
+### Clustering redux
+
+These data don't have clusters assigned yet, so let's add those, using the PCA matrix we have for the gene expression data.
+
+
+```{r}
+# perform clustering
+nn_k <- 20
+nn_clusters <- bluster::clusterRows(
+  reducedDim(sce, "PCA"), 
+  bluster::NNGraphParam(
+    k = nn_k, 
+    type = "jaccard",
+    cluster.fun = "louvain"
+  )
+)
+
+# add clusters to colData
+sce$nn_cluster <- nn_clusters
+```
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "nn_cluster")
+```
+But what are these clusters? Do they correspond to cell types?
+
+## Cell surface markers
+
+The markers we have were presumably chosen for the fact that they mark particular cell types, so let's look at a few.
+
+CD3 is a marker of T-cells 
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD3")
+```
+
+CD4+ T-cells
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD4")
+```
+CD8+ T-cells
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD8")
+```
+
+Looking at all of the markers here, but we probably don't want to do this, as we haven't introduced `purrr` yet.
+```{r}
+all_plots <- adt_tags %>%
+  purrr::map(~ scater::plotUMAP(sce, colour_by = .))
+
+patchwork::wrap_plots(all_plots)
```

Comment:
TIL! this is cool, although yeah maybe not for this notebook. It could be an _advanced_ exercise option?